source/scripting/Entity.cs

```diff
@@ -561,6 +561,22 @@ public bool HasBeenDamagedBy(Entity entity)
 		{
 			return Function.Call<bool>(Hash.HAS_ENTITY_BEEN_DAMAGED_BY_ENTITY, Handle, entity.Handle, 1);
 		}
+		public bool HasBeenDamagedBy(WeaponHash weapon)
```

Comment:
I decided to keep it as HasBeenDamagedBy to match the function above - HasBeenDamagedBy(Entity);